src/cmake/boost-fallback/boost-std-configuration.h.cmake

```diff
@@ -0,0 +1,94 @@
+#ifndef BOOST_STD_CONFIGURATION_H
+#define BOOST_STD_CONFIGURATION_H
+
+#cmakedefine01 STD_THREAD_FOUND
+#cmakedefine01 Boost_THREAD_FOUND
+#cmakedefine01 STD_MUTEX_FOUND
+#cmakedefine01 Boost_MUTEX_FOUND
+#cmakedefine01 STD_SHARED_MUTEX_FOUND
+#cmakedefine01 Boost_SHARED_MUTEX_FOUND
+#cmakedefine01 STD_SHARED_PTR_FOUND
+#cmakedefine01 Boost_SHARED_PTR_FOUND
+#cmakedefine01 STD_ATOMIC_FOUND
+#cmakedefine01 Boost_ATOMIC_FOUND
+
+#if STD_THREAD_FOUND
+#include <thread>
+namespace ${NAMESPACE_ALIAS} {
+    typedef std::thread thread;
+}
+#elif Boost_THREAD_FOUND
+#include <boost/thread.hpp>
+namespace ${NAMESPACE_ALIAS} {
+    typedef boost::thread thread;
+}
+#endif
+
+#if STD_MUTEX_FOUND
+#include <mutex>
+#include <condition_variable>
+namespace ${NAMESPACE_ALIAS} {
+    typedef std::mutex mutex;
+    template <typename T>
+    using unique_lock = std::unique_lock<T>;
```

Comment:
This syntax doesn't work for C++ less than 11, I'm running into this with my legacy C++-Builder 10.2 UPD 3 with 32 Bit BCC. So simply don't support those old compilers anymore? On the mailing list we had a discussion in the past about different versions of the lib or not etc.

https://stackoverflow.com/a/2795024/2055163

Reply:
Using inheritance compiles, but one needs to provide all CTORs, operators etc. manually this way:

```cpp
template <typename T>
class unique_lock : public boost::unique_lock<T>
{
	public:
		unique_lock(const T& mutex) : boost::unique_lock(mutex) {}
};
```

Reply:
[Only at least C++11 will be supported](https://lists.apache.org/thread.html/re744feb89b0bf9a15b03d145a88fe0d0be48ad069d12a31b8ba5d461%40%3Cdev.logging.apache.org%3E) in future and especially `BCC32` is a dead-end for everyone still using it anyway. Embarcadero prefers its CLANG-based alternatives as well for some years now, people need to upgrade their projects.